docs/Technical-Documentation/tech-memos/login-gov-set-integration/login-gov-set-integration.md

```diff
@@ -0,0 +1,359 @@
+# Login.gov Security Event Token (SET) Integration
+
+**Audience**: TDP Software Engineers <br>
+**Subject**: Integration of Login.gov Security Event Tokens with Account Recreation Handling <br>
+**Date**: July 17, 2025 <br>
+
+## Summary
+This technical memorandum outlines the implementation plan for integrating [Security Event Tokens (SETs)](https://developers.login.gov/security-events/) from Login.gov into the TDP application. The integration will enable TDP to:
+1. Handle security-related notifications about user accounts
+2. Automatically manage account recreation scenarios when users delete and recreate their Login.gov accounts
+3. Maintain audit history and respond appropriately to security events
+
+## Background
+Login.gov provides a Security Event Token (SET) notification system that follows the OpenID RISC Event Types specification. These notifications are sent as JWT tokens via HTTP POST requests when specific security events occur for users, such as:
+- Account disabling
+- Account purging (when users delete their accounts)
+- Password resets
+- Recovery information changes
+
+A common issue in production is that users sometimes delete their Login.gov accounts and recreate them, which results in a new `sub` claim for the same email address. This causes authentication issues in TDP. TDP manages users by the `sub`/`login_gov_uuid`. That is, TDP looks up users via the `login_gov_uuid`. When a user's ID changes, TDP tries to create a new account. However, the account creation will fail because TDP places a unique constraint on the user's `email`. Since the user's email hasn't changed TDP's ORM throws a unique key violation error, which is presented as a plain old json object to the user (seen below). By implementing SET handling, particularly for the "Account Purged" event, we can automatically manage these account recreation scenarios without requiring additional user verification steps.
```

Comment:
One question: if user creates a new account with same email, why can't we just upsert that new account?

Reply:
If we are 100% positive with nonce and token that the user information is coming from cloud.gov, then I don't see any point in not considering user email as reference, unless I am missing something